### PR TITLE
Feature/execution artificats

### PR DESCRIPTION
**Pull request by sudarshan0212**
Thank you for your contribution.
Before submitting this PR, please make sure:

- [ ] PR description and commit message should describe the changes done in this PR
- [ ] Verify the PR is pointing to correct branch i.e. Release or Beta branch if the code fix is for specific release , else point it to master
- [ ] Latest Code from master or specific release branch is merged to your branch
- [ ] No unwanted\commented\junk code is included
- [ ] No new warning upon build solution
- [ ] Code Summary\Comments are added to my code which explains what my code is doing
- [ ] Existing unit test cases are passed
- [ ] New Unit tests are added for your development
- [ ] Sanity Tests are successfully executed for New and Existing Functionality
- [ ] Verify that changes are compatible with all relevant browsers and platforms.
- [ ] After creating pull request there should not be any conflicts
- [ ] Resolve all Codacy comments
- [ ] Builds and checks are passed before PR is sent for review
- [ ] Resolve code review comments
- [ ] Update the Help Library document to match any feature changes

### CODE REVIEW

File: Ginger/Ginger/Actions/UserControls/UCArtifact.xaml.cs [hunk 1]
```diff
@@ -0,0 +1,192 @@
...
+            string tempFilePath = GingerCoreNET.GeneralLib.General.CreateTempTextFile(ArtifactPath);
+            if (System.IO.File.Exists(tempFilePath))
+            {
+                DocumentEditorPage docPage = new DocumentEditorPage(ArtifactPath, enableEdit: false, UCTextEditorTitle: string.Empty);
+                docPage.Width = 800;
+                docPage.Height = 800;
+                docPage.ShowAsWindow(ArtifactName);                
+                return;
+            }          
+            Reporter.ToUser(eUserMsgKey.StaticErrorMessage, "Failed to load raw response view, see log for details.");
```
Comment (MeniKadosh1) [inline Ginger/Ginger/Actions/UserControls/UCArtifact.xaml.cs:96]: error not seems to be relevant

File: Ginger/GingerCoreNET/Run/RunListenerLib/CenteralizedExecutionLogger/AccountReportApiHandler.cs [hunk 2]
```diff
@@ -363,6 +364,81 @@ public async Task UploadImageAsync(Guid executionId, List<string> filePaths)
...
+                {
+                    try
+                    {
+                        Reporter.ToStatus(eStatusMsgKey.PublishingToCentralDB, null, "Uploading execution artifacts to central DB");
+
+                        long fileSize = 0;
+
+                        List<string> temp = new List<string>();
+
+                        foreach (string screenshot in filePaths)
```
Comment (MeniKadosh1) [inline Ginger/GingerCoreNET/Run/RunListenerLib/CenteralizedExecutionLogger/AccountReportApiHandler.cs:381]: it's artifacts but code params are still "screenshot" - need to correct it

File: Ginger/GingerCoreNET/Run/GingerExecutionEngine.cs [hunk 3]
```diff
@@ -2348,6 +2348,14 @@ private void ProcessScreenShot(Act act, GingerRunner.eActionExecutorType ActionE
             }
         }
 
+        private void SaveArtifactsScreenshot(Act act, string screenshot)
+        {
+            act.ScreenShots.Add(screenshot);
+            if (act.GetType() == typeof(ActVisualTesting))
```
Comment (MeniKadosh1) [inline Ginger/GingerCoreNET/Run/GingerExecutionEngine.cs:2354]: can't we catch it from the ActVisualTesting or from the Drivers somehow instead placing the IF it in generic place

**Review by MeniKadosh1 — APPROVED** [review]
Approving but needs to fix the comments as part of bugs fixing

File: Ginger/Ginger/Actions/UserControls/UCArtifact.xaml.cs (hunk 1 above)
Comment (sudarshan0212) [reply Ginger/Ginger/Actions/UserControls/UCArtifact.xaml.cs]: Done.

File: Ginger/GingerCoreNET/Run/GingerExecutionEngine.cs (hunk 3 above)
Comment (sudarshan0212) [reply Ginger/GingerCoreNET/Run/GingerExecutionEngine.cs:2353]: Done

File: Ginger/GingerCoreNET/Run/RunListenerLib/CenteralizedExecutionLogger/AccountReportApiHandler.cs (hunk 2 above)
Comment (sudarshan0212) [reply Ginger/GingerCoreNET/Run/RunListenerLib/CenteralizedExecutionLogger/AccountReportApiHandler.cs]: Done.

**Pull request merged**